### ISSUE TITLE
Remove client-side instance blocking

### ISSUE THREAD

**Issue by IAmCaligvla** [LucasGGamerM/moshidon#184]
According to this [changelog](https://github.com/sk22/megalodon/releases/tag/v1.1.5%2Bfork.68) a while ago Megalodon implemented a client-side instance block for many servers the creator personally dislikes, Moshidon unfortunately seems to have inherented this. 

Reverting this arbitrary change would be nice for the end user if possible.

**Comment by LucasGGamerM** [comment]
As the official mastodon client does not have it, I believe it would be safe to just remove the list

**Comment by FineFindus** [comment]
If I view this [block list](https://github.com/sk22/megalodon/blob/main/mastodon/src/main/assets/blocks.tsv) correctly, many of the servers are blocked because of harassment or even CSAM. So while it is technically possible, just for ethical reason (and possible legal ones, as Google might decide to remove the app for it), I would highly recommend against removing this!

**Comment by LucasGGamerM** [comment]
Yeah, I took a better look at this a while back and I agree on the fact that the list should be kept

**Comment by Taullo** [comment]
Is this going to stay disabled? I'm all for blocking hate speech, actual CSAM, and hate speech but there's instances on there (including my own) that got blocked for simply hosting fictional kink content. We were thinking about maintaining a fork of Megalodon that simply removes the instance blocking code, but it'd be nice to recommend this instead.

**Comment by FineFindus** [comment]
I would strongly prefer it if the block list stays included.

To get your server removed, I think it would be best if you either opened a pr against Megalodon, or even better talk directly to the creators of the list (the list is based on one which multiple server use).

I don't think @sk22 would be happy about you distributing an app version without blocking.

**Comment by Taullo** [comment]
> I would strongly prefer it if the block list stays included.
> 
> To get your server removed, I think it would be best if you either opened a pr against Megalodon, or even better talk directly to the creators of the list (the list is based on one which multiple server use).
> 
> I don't think @sk22 would be happy about you distributing an app version without blocking.

I've already commented upstream and contacted the block list author. However, it's my assumption that if they've included the list, they've vetted it and taken umbrage with the content we have hosted. This fork wasn't the original author of the commit that included the blocklist so I think stating my concerns about upstream here are perfectly valid.

Respectfully, the code is GPL-3 and based on the original Mastodon app. It's not their code, or anyone else's. GPL-3 allows us to fork the code and make specific changes to it. Frankly, I think that blocking harmless instances under the guise of "stopping nazis" (upstream and the block list author's word) is incredibly user hostile. Again, respectfully, if sk22 doesn't want the code to be forked to remove what we think is a harmful commit, then they shouldn't have used someone else's code that was licensed under a free software license.

**Comment by FineFindus** [comment]
> Respectfully, the code is GPL-3 and based on the original Mastodon app. It's not their code, or anyone else's. GPL-3 allows us to fork the code and make specific changes to it.

I didn't mean to imply that you weren't allowed to. There is a difference between being allowed to and going against the developer's intentions. Also, technically everything they wrote is their code, and they are free to re-license it in the future if they wish. But I don't think discussing licensing is going to move this discussion forward.

**Comment by Taullo** [comment]
> > Respectfully, the code is GPL-3 and based on the original Mastodon app. It's not their code, or anyone else's. GPL-3 allows us to fork the code and make specific changes to it.
> 
> I didn't mean to imply that you weren't allowed to. There is a difference between being allowed to and going against the developer's intentions. Also, technically everything they wrote is their code, and they are free to re-license it in the future if they wish. But I don't think discussing licensing is going to move this discussion forward.

I don't think it was the Mastodon Project's intention for their code to be used to splinter the fediverse based on a list generated from a random website but that's what happened to begin with.  The fediverse is supposed to be an open network of websites that communicate with one another and a developer writing code to remove that is flying in the face of what the original code was intended for. 

Also that's not how GPL works. It can be re-licensed under additional licenses, but it cannot be removed from being GPL in the future. Anyone can take any GPL-licensed code they want, modify it, and release it under GPL. That's how open source software works. Especially in this case where the source project is GPL and the person we're talking about is modifying someone else's GPL code. I'm not sure why you even attempted to @ the original developer to warn them of someone else modifying their code when placed in an open source project, but here we are.

**Comment by FineFindus** [comment]
> Also that's not how GPL works

You're probably right, I'm sorry. I haven't slept very much, so my brain does not really function.

>  I'm not sure why you even attempted to @ the upstream developer to warn them of someone else modifying their open source GPL code, but here we are.

I did not ping them to warn them, but to bring them into the discussion, as I felt that their opinion had value in this discussion, after all they were the one who introduced the feature in the first place. But since you seem to assume the worst of me, I'll refrain from further participation. Good luck with your fork!

**Comment by Taullo** [comment]
> I did not ping them to warn them, but to bring them into the discussion, as I felt that their opinion had value in this discussion, after all they were the one who introduced the feature in the first place. But since you seem to assume the worst of me, I'll refrain from further participation. Good luck with your fork!
You and I both know that's not what you were doing and instead wanted help reprimanding someone for even suggesting a fork of a GPL project. Good night.

**Comment by LucasGGamerM** [comment]
I talked a bit with @FineFindus in private and he provided me with the following screenshot of posts on your server: 
![image](https://github.com/LucasGGamerM/moshidon/assets/71328265/a0fa84ad-2406-4415-985d-79e662cdcb5b)
What is your stance on this kind of content relating to underage type of sexual content?

**Comment by FullyAutomaticWitchfire** [comment]
As an active user on Taullo's instance since its inception(@ Satanist), I can confirm that actual abuse is taken extremely seriously, and action is taken quickly and swiftly whenever a report is made.

The instance solely hosts fictional content, and in the case of underage fictional content, has been scientifically proven by sexologists to not lead to real life action, which is the sole reason it is still legal in places like Denmark, as the Danish government commissioned a report on it and got told this in return.

If you can manage to find CSAM on the instance, you should absolutely report it to it can be taken down. Equating fictional content to real abuse emboldens abusers, makes real abuse harder to report and hurts real life victims.

**Comment by rudytues-day** [comment]
as an active aethy user and csam survivor i would like to add that the love and support i have found on aethy is second to none in any online communities i've ever been a part of in my entire life. additionally, any reports that i've made in the six months i've been using fedi were regarding accounts outside of our instance and handled by admin immediately.

i get it if you think it's gross—it's definitely not for the weak—but our community has made an active effort to combat predatory behavior and real abuse of minors, which is very, very different than pictures you don't like.

**Comment by sk22** [comment]
did you (@Taullo) hear back from seirdy (the fedinuke maintainer)? i'll definitely like to stick with the fedinuke list and not make exceptions; maintaining a modified version of the fedinuke list is outside the scope of my project and also outside of what i'm willing to commit to. (what the issue title is concerned: i also definitely don't want to remove the blocklist - i think that it's understandable that i'd like to keep fascists (et al) from using my app)

**Comment by FluffyHyena** [comment]
> I talked a bit with @FineFindus in private and he provided me with the following screenshot of posts on your server: ![image](https://user-images.githubusercontent.com/71328265/251513179-a0fa84ad-2406-4415-985d-79e662cdcb5b.png) What is your stance on this kind of content relating to underage type of sexual content?

I would argue that in sharing screenshots in public without making an effort to conceal usernames you are doing much more real harm to actual people that any properly tagged, filtered and warned for at-most-risqué sketch of a cartoon character from a mainstream show aimed at adults where children often engage in sexual relationships between themselves and/or older people. You know, a little show called _South Park. Which features this content often. Onscreen._

**Comment by LucasGGamerM** [comment]
I have redacted the screenshots. I apologize for mistakenly sharing them without making an effort to censor usernames. 

In any case, I unblocking your instance might be on the table. Tomorrow I will talk to the fedinuke guys about the list to better understand the reason you guys were put there in the first place. And after doing so, I will see what can be done

**Comment by EeveeEuphoria** [comment]
Thank you very much for taking this all into consideration, I've started using this app recently, and would like to continue to use it assuming aethy would be removed from the list. My main instance works with it, but tusky let's me have both instances on one app, it'd be nice to use both on here!

**Comment by MylesGit** [comment]
Can this issue be marked resolved?

**Comment by shuvashish76** [comment]
Upstream issue : https://github.com/sk22/megalodon/issues/511
Reasons mentioned in this blog post : https://seirdy.one/posts/2023/05/02/fediverse-blocklists/

**Comment by cirnoe** [comment]
anyone maintaining a fork without this ridiculous measure?
